strawberryfields/configuration.py

```diff
@@ -0,0 +1,189 @@
+load, store, save, and modify configuration options for Strawberry Fields.
+
+Behaviour
+--------
+
+On first import, Strawberry Fields attempts to load the configuration file `config.toml`, by
+scanning the following three directories in order of preference:
+
+1. The current directory
+2. The path stored in the environment variable ``SF_CONF``
+3. The default user configuration directory:
+
+   * On Linux: ``~/.config/strawberryfields``
+   * On Windows: ``~C:\Users\USERNAME\AppData\Local\Xanadu\strawberryfields``
+   * On MacOS: ``~/Library/Preferences/strawberryfields``
+
+If no configuration file is found, a warning message will be displayed in the logs,
+and all device parameters will need to be passed as keyword arguments when
+loading the device.
+
+The user can access the initialized configuration via `strawberryfields.config`, view the
+loaded configuration filepath, print the configurations options, access and modify
+them via keys, and save/load new configuration files.
+
+Configuration files
+-------------------
+
+The configuration file `config.toml` uses the `TOML standard <https://github.com/toml-lang/toml>`_,
+and has the following format:
+
+.. code-block:: toml
+
+    [api]
+    # Options for the Strawberry Fields Cloud API
+    authentatication_token = "071cdcce-9241-4965-93af-4a4dbc739135"
+    hostname = "localhost"
+    use_ssl = true
+
+Summary of options
+------------------
+
+todo
+
+Summary of methods
+------------------
+
+.. currentmodule:: strawberryfields.configuration.Configuration
+
+.. autosummary::
+    path
+    load
+    save
+
+Code details
+~~~~~~~~~~~~
+
+.. currentmodule:: strawberryfields.configuration
+
+"""
+import os
+import logging as log
+
+import toml
+from appdirs import user_config_dir
+
+log.getLogger()
+
+
+DEFAULT_CONFIG = {"api": {"authentatication_token": "", "hostname": "localhost", "use_ssl": True}}
+
+
+class ConfigurationError(Exception):
+    """Exception used for configuration errors"""
+
+
+class Configuration:
+    """Configuration class.
+
+    This class is responsible for loading, saving, and storing StrawberryFields
+    and plugin/device configurations.
+
+    Args:
+        name (str): filename of the configuration file.
+        This should be a valid TOML file. You may also pass an absolute
+        or a relative file path to the configuration file.
+    """
+
+    def __str__(self):
+        return "{}".format(self._config)
+
+    def __repr__(self):
+        return "Strawberry Fields Configuration <{}>".format(self._filepath)
+
+    def __init__(self, name="config.toml"):
+        # Look for an existing configuration file
+        self._config = DEFAULT_CONFIG
+        self._config_file = {}
+        self._filepath = None
+        self._name = name
+        self._user_config_dir = user_config_dir("strawberryfields", "Xanadu")
+        self._env_config_dir = os.environ.get("SF_CONF", "")
+
+        # search the current directory, the directory under environment
```

Comment:
Various occurrences of this, but PEP8 suggests that the first word should always be capitalized in comments (https://www.python.org/dev/peps/pep-0008/#id30)

Reply:
Done